inc/namespace.php

```diff
@@ -53,6 +54,22 @@ function maybe_disable_password_reset( $allow, $user_id ) {
 	return $allow;
 }
 
+/**
+ * Prevent users with disabled accounts to use application passwords.
+ *
+ * @param bool $allow Whether to enable application passwords for the user. Default true.
+ * @param WP_User $user user that is trying to access application passwords.
+ *
+ * @return bool true if application passwords should be enabled, false if it should be disabled.
+ */
+function maybe_disable_application_passwords_for_user( $allow, $user ) {
+	if ( is_disabled( $user ) ) {
+		$allow = false;
```

Comment:
Just to avoid a little indirection:

```suggestion
		return false;
```